addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/discovery/NikoHomeControlDiscoveryService.java

```diff
@@ -82,32 +83,50 @@ public void discoverDevices() {
             switch (nhcAction.getType()) {
                 case 0: // handles all-off
                 case 1: // switch
-                    addDevice(new ThingUID(THING_TYPE_ON_OFF_LIGHT, this.handler.getThing().getUID(),
+                    addActionDevice(new ThingUID(THING_TYPE_ON_OFF_LIGHT, this.handler.getThing().getUID(),
                             Integer.toString(actionId)), actionId, thingName, thingLocation);
 
                     break;
                 case 2: // dimmer
-                    addDevice(new ThingUID(THING_TYPE_DIMMABLE_LIGHT, this.handler.getThing().getUID(),
+                    addActionDevice(new ThingUID(THING_TYPE_DIMMABLE_LIGHT, this.handler.getThing().getUID(),
                             Integer.toString(actionId)), actionId, thingName, thingLocation);
                     break;
                 case 4: // rollershutter
                 case 5:
-                    addDevice(new ThingUID(THING_TYPE_BLIND, this.handler.getThing().getUID(),
+                    addActionDevice(new ThingUID(THING_TYPE_BLIND, this.handler.getThing().getUID(),
                             Integer.toString(actionId)), actionId, thingName, thingLocation);
                     break;
                 default:
                     logger.debug("Niko Home Control: unrecognized action type {} for {} {}", nhcAction.getType(),
                             actionId, thingName);
             }
         }
+
+        Map<Integer, NhcThermostat> thermostats = nhcComm.getThermostats();
+
+        for (Map.Entry<Integer, NhcThermostat> thermostat : thermostats.entrySet()) {
```

Comment:
Maybe an `thermostatEntry `would be more suitable because the `thermostat.getValue()` is the actual `thermostat `